docs/design/training-pre-check.md

```diff
@@ -0,0 +1,121 @@
+# Training Pre Check Procedure Introduction
+
+## Background
+
+As the scale of training production expands, the factors that may cause 
+runtime exceptions related to the environment during training are gradually 
+increasing. Although the current DLRover adds a so-called node-check before 
+training (which runs a small-scale training task to detect any abnormal 
+scenarios), it primarily observes availability from the training dimension 
+perspective. Therefore, it is not easy to specifically diagnose problematic 
+nodes and classify issues. 
+
+So we are seeking a more advanced and precise check, 
+conducting certain related inspections right when all nodes are in place 
+(or not yet in place) to identify issues as early as possible and implement 
+fault tolerance or blockages in advance. The checks mentioned earlier rely more 
+on external services, as they require specialized and more precise validation 
+of components such as containers and networks. Traditional training frameworks 
+do not and should not involve these components. 
+
+Considering that the actual 
```

Comment:
It'd better not emphasize "no actual implementation" here since the implementation is expected to come later. It is ok to describe the interface here.

Reply:
done